docs(logging): log4j2 support updates (#11254)
strimzi/strimzi-kafka-operator: documentation/api/io.strimzi.api.kafka.model.connect.KafkaConnectSpec.adoc

= Logging
Kafka Connect has its own configurable loggers:

> * `rootLogger.level`
> * `logger.org.reflections`

Reviewer: ```suggestion
* `rootLogger.*`
* `logger.reflections.name = org.reflections`
  `logger.reflections.*`
```

= Logging
Kafka Connect has its own configurable loggers:

* `rootLogger.*`
* `logger.reflections.name = org.reflections`

> * `logger.reflections.*`

Author: Thanks @scholzj. Before I proceed, I just want to confirm how you see the presentation. Like this?

> Kafka Connect has its own configurable loggers:
> 
> * `rootLogger`
> * `reflections`

As an alternative to adding the log4j1 examples back, we don't want to add a note in the relevant logging sections of the API reference to say something like: this is for Kafka 4/log4j2, for Kafka 3.9/log4j1 go look at 0.45 doc?

Assuming, with the addition of a few comments (as suggested), we can keep the deploying content as is

> * `logger.reflections.*`

Reviewer 2: Yeah, I thikn just mentioning the Log4j1 and linking to 0.45.0 docs is fine.

curl -s http://<connect-cluster-name>-connect-api:8083/admin/loggers/
----

Kafka Connect uses the Apache `log4j2` logger implementation.
Use the `logging` property to configure loggers and logger levels.

> NOTE: For Strimzi 0.45 and earlier, log4j1 was used for Kafka components. 
> Refer to the Strimzi 0.45 documentation for loggers and logging configuration examples based on log4j1.

Reviewer 2: This is not a question of Strimzi version but Kafka version. So this needs ot be reworked here as well as in the other places. E.g.:

```suggestion
NOTE: Kafka 3.9 and earlier uses Log4j1 got logging. 
Refer to the Strimzi 0.45 documentation for loggers and logging configuration examples based on log4j1 that you can use with Strimzi 0.46 and Kafka 3.9.x: <LINK-TO-THE-DOCS>
```

**We should probably also mark it as IMPORTANT and not NOTE and move it to the top of the section. Ir is pretty lost here in the middle and for users it makes zero sense to start reading this chapter if they use Kafka 3.9 as none of it is valid for them.**

curl -s http://<connect-cluster-name>-connect-api:8083/admin/loggers/
----

> Kafka Connect uses the Apache `log4j2` logger implementation.

Author: I've added note, as discussed.

  logging:
    type: inline
    loggers:
      rootLogger.level: INFO

>       logger.org.apache.kafka.connect.runtime.WorkerSourceTask: TRACE
>       logger.org.apache.kafka.connect.runtime.WorkerSinkTask: DEBUG

Reviewer: ```suggestion
      logger.sourcetask.name = org.apache.kafka.connect.runtime.WorkerSourceTask
      logger.sourcetask.level = TRACE
      logger.sinktask.name = org.apache.kafka.connect.runtime.WorkerSinkTask
      logger.sinktask.level = DEBUG
```
